### PR TITLE
Feature#7 frontend index

### LINKED ISSUE #7: Frontend index
# REQUIREMENTS
make the design of the mockup into the home page.

# DESIGN
image

### PR DESCRIPTION
**Pull request by AngelBFdev**
# Description
- Header design 
- Form design and layout for the Index page
- Clock design 
- Clock functionality with javascript

## Screenshots
Mockup
image
Frontend
image

### CODE REVIEW

**Comment by kevinnio** [comment]
I think it should be `Check In` and `Check Out`.

File: Gemfile [hunk 1]
```diff
@@ -38,7 +40,7 @@ group :development do
   gem 'web-console', '>= 4.1.0'
   # Display performance information such as SQL time and flame graphs for each request in your browser.
   # Can be configured to work on production as well see: https://github.com/MiniProfiler/rack-mini-profiler/blob/master/README.md
-  gem 'rack-mini-profiler', '~> 2.0'
+  # gem 'rack-mini-profiler', '~> 2.0's
```
Comment (kevinnio) [inline Gemfile]: Did you comment this line out because it was failing? Looks like that `s` at the end might have been the culprit. 🤔

File: Gemfile [hunk 2]
```diff
@@ -3,6 +3,8 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby '3.0.2'
 
+gem 'wdm', '>= 0.1.0'
```
Comment (kevinnio) [inline Gemfile:6]: Is this https://rubygems.org/gems/wdm? Why did you need to add a low-level gem like that to the project? 🤔

File: Gemfile.lock [hunk 3]
```diff
@@ -201,6 +207,7 @@ GEM
     zeitwerk (2.5.1)
 
 PLATFORMS
+  x64-mingw32
```
Comment (kevinnio) [inline Gemfile.lock:210]: ...ohh, maybe `wdm` was added because someone is using `mingw` to develop and something was broken? 🤔

File: app/assets/stylesheets/src/base/global.css.scss [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+html {
+  background-color: $background_color;
+  box-sizing: border-box;
+  font-family: $font;
+  font-size: 62.5%;
+  scroll-padding-top: 0rem;
+}
+
+*,
```
Comment (kevinnio) [inline app/assets/stylesheets/src/base/global.css.scss:11]: Is this style required? I think `normalize.css` already includes a style similar to this.

File: app/assets/stylesheets/src/base/normalize.css.scss [hunk 5]
```diff
@@ -0,0 +1,349 @@
+/*! normalize.css v8.0.1 | MIT License | github.com/necolas/normalize.css */
```
Comment (kevinnio) [inline app/assets/stylesheets/src/base/normalize.css.scss]: Including this stylesheet is ok, but usually, we add the `.min` version of third-party stylesheets like this one, since they're smaller and faster to download for users.

File: app/javascript/packs/clock.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+        hr = 12;
+        am_pm = "AM";
+    }
+
+    hour = hour < 10 ? "0" + hour : hour;
+    min = min < 10 ? "0" + min : min;
+    sec = sec < 10 ? "0" + sec : sec;
+
+    let currentTime = hour + ":" + min;
+    let meriedem = am_pm;
```
Comment (kevinnio) [inline app/javascript/packs/clock.js]: I get `currentTime` but looks like `meriedem` and `seconds` are not really useful variables here. 🤔

File: app/javascript/stylesheets/application.scss [hunk 7]
```diff
@@ -1 +1 @@
-@import 'bootstrap'
+@import 'bootstrap';
```
Comment (kevinnio) [inline app/javascript/stylesheets/application.scss:1]: Oh, just noticed you're importing `bootstrap` here. As far as I'm aware, Bootstrap includes its own version of `normalize.css` with it, so maybe adding the stylesheet directly is not really needed? 🤔

File: package.json [hunk 8]
```diff
@@ -10,11 +10,11 @@
     "bootstrap": "^5.1.3",
     "popper.js": "^1.16.1",
     "turbolinks": "^5.2.0",
-    "webpack": "^4.46.0",
-    "webpack-cli": "^3.3.12"
+    "webpack": "4.46.0",
```
Comment (kevinnio) [inline package.json:13]: Any particular reason to change the version constraints of these packages? 🤔

**Review by kevinnio — COMMENTED** [review]
Left some questions and suggestions.

**Review by YilKanda — COMMENTED** [review]
My comment is directed regarding your commits:

1. I notice that you have a commit that says -in process- and one that says -complete-. It is not recommended to do this, since it remembers that your commits must be significant, that is, each commit contributed to a change within your project.

imagen

2. Try to be more specific in your commits, that is, what change was made? Why was it done? How does this change modify the direction of the project? Do not assume the reviewer understands what the original problem was, ensure you add it. Do not assume your code is self-explanatory.

imagen

Here's a great template of a good commit message originally written by Tim pope:

```
Capitalized, short (50 chars or less) summary

More detailed explanatory text, if necessary.  Wrap it to about 72
characters or so.  In some contexts, the first line is treated as the
subject of an email and the rest of the text as the body.  The blank
line separating the summary from the body is critical (unless you omit
the body entirely); tools like rebase can get confused if you run the
two together.

Write your commit message in the imperative: "Fix bug" and not "Fixed bug"
or "Fixes bug."  This convention matches up with commit messages generated
by commands like git merge and git revert.

Further paragraphs come after blank lines.

- Bullet points are okay, too

- Typically a hyphen or asterisk is used for the bullet, followed by a
  single space, with blank lines in between, but conventions vary here

- Use a hanging indent

If you use an issue tracker, add a reference(s) to them at the bottom,
like so:

Resolves: #123
```

3. Specify the type of commit:


    **feat:** The new feature you're adding to a particular application
    **fix:** A bug fix
    **style:** Feature and updates related to styling
    **refactor:** Refactoring a specific section of the codebase
    **test:** Everything related to testing
    **docs:** Everything related to documentation
    **chore:** Regular code maintenance.[ You can also use emojis to represent commit types]

File: package.json (hunk 8 above)
Comment (MauricioNC) [reply package.json:13]: We got an error when ran the app, the error shows something like webpacker compilation is not configured as true, and that we need to install weback, I do not remember the error specifically, but we found as a solution run the command: rails webpacker:install

File: Gemfile (hunk 1 above)
Comment (MauricioNC) [reply Gemfile]: mmm, I don't know why this line is commented, no one us commented this line but when we ran the app we had no problemas. We will search more about this gem.

File: Gemfile (hunk 2 above)
Comment (MauricioNC) [reply Gemfile:6]: When we tryied to make the base project we got some issues at the momment to run the rake db:migrate command, searching on internet we foun that installing this gem the problem it would be solved and yes, now we deleted this gem and ran the app and everything is okay... so, I don't know what was the problem initially🤔

File: Gemfile.lock (hunk 3 above)
Comment (MauricioNC) [reply Gemfile.lock:210]: We are developing this project on windows, and yeah, I am using mingw, maybe that was the problem with the database creation, but now that we delete this gem, we have no errors, I don't know if it's necessary have installed this gem for future possible errors

File: app/javascript/packs/clock.js (hunk 6 above)
Comment (AngelBFdev) [reply app/javascript/packs/clock.js]: You're right, we are working on it. Thanks

File: app/javascript/stylesheets/application.scss (hunk 7 above)
Comment (MauricioNC) [reply app/javascript/stylesheets/application.scss:1]: Oooh yeah, you're right, we don't need our normalize.css

File: app/assets/stylesheets/src/base/normalize.css.scss (hunk 5 above)
Comment (MauricioNC) [reply app/assets/stylesheets/src/base/normalize.css.scss]: This is resolved with normalize.css from bootstrap

File: app/assets/stylesheets/src/base/global.css.scss (hunk 4 above)
Comment (MauricioNC) [reply app/assets/stylesheets/src/base/global.css.scss:11]: This is resolved with normalize.css from bootstrap

File: Gemfile (hunk 2 above)
Comment (kevinnio) [reply Gemfile:6]: I see. Well, if the app works just as well without it, then let's remove it. The fewer the gems, the better! Helps to keep [technical debt](https://www.productplan.com/glossary/technical-debt/) down.

File: Gemfile (hunk 1 above)
Comment (kevinnio) [reply Gemfile]: This gem is not required for the app to work, but it can be helpful while developing it. I'll let you guys decide if you want to keep it or not, so if you decide to remove it, please remove the line completely instead of just commenting it out.

**Review by kevinnio — DISMISSED** [review]

File: app/assets/stylesheets/src/base/variables.css.scss [hunk 9]
```diff
@@ -0,0 +1,10 @@
+
+//custom
+$background_color: #252525;
+$background_table: #333333;
+$font_color: #FFFFFF;
+$heading_color: #181818;
+$secondary_color: #00E8B0;
+$table_cell: #C4C4C4;
+//font family
+$font: Consolas,monaco,monospace;
```
Comment (JuanequeX) [inline app/assets/stylesheets/src/base/variables.css.scss]: Did can add a blank line here

File: app/assets/stylesheets/src/base/variables.css.scss [hunk 10]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (JuanequeX) [inline app/assets/stylesheets/src/base/variables.css.scss:1]: Delete this extra space

File: app/assets/stylesheets/src/layout/form.css.scss [hunk 11]
```diff
@@ -0,0 +1,78 @@
+@import "../base/variables.css";
+
+.main {
+
```
Comment (JuanequeX) [inline app/assets/stylesheets/src/layout/form.css.scss]: Delete this extra space

File: app/assets/stylesheets/src/layout/header.css.scss [hunk 12]
```diff
@@ -0,0 +1,29 @@
+@import '../base/variables.css';
+
+
+.navbar {
+
```
Comment (JuanequeX) [inline app/assets/stylesheets/src/layout/header.css.scss]: Delete this extra space

File: app/javascript/packs/clock.js [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+    min = min < 10 ? "0" + min : min;
+    sec = sec < 10 ? "0" + sec : sec;
+
+    let currentTime = hour + ":" + min;
+
+    document.getElementById("hour").innerHTML = currentTime;
+    document.getElementById("meriedem").innerHTML = am_pm;
+    document.getElementById("seconds").innerHTML = sec;
+}
+showTime();
```
Comment (JuanequeX) [inline app/javascript/packs/clock.js]: Add a blank line here

**Review by JuanequeX — CHANGES_REQUESTED** [review]

**Review by kevinnio — DISMISSED** [review]

**Review by YilKanda — APPROVED** [review]
Check The indentation of your files.

**Review by JuanequeX — APPROVED** [review]
LGTM

File: app/assets/stylesheets/src/base/variables.css.scss (hunk 9 above)
Comment (MauricioNC) [reply app/assets/stylesheets/src/base/variables.css.scss]: Done.

File: app/assets/stylesheets/src/base/variables.css.scss (hunk 10 above)
Comment (MauricioNC) [reply app/assets/stylesheets/src/base/variables.css.scss:1]: Done.

File: app/assets/stylesheets/src/layout/form.css.scss (hunk 11 above)
Comment (MauricioNC) [reply app/assets/stylesheets/src/layout/form.css.scss]: Done.

File: app/assets/stylesheets/src/layout/header.css.scss (hunk 12 above)
Comment (MauricioNC) [reply app/assets/stylesheets/src/layout/header.css.scss]: Done.

File: app/javascript/packs/clock.js (hunk 13 above)
Comment (MauricioNC) [reply app/javascript/packs/clock.js]: Done.

**Review by kevinnio — APPROVED** [review]
